networks/xml_cnn.py

```diff
@@ -0,0 +1,74 @@
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from networks.base import BaseModel
+
+
+def out_size(l_in, kernel_size, channels, padding=0, dilation=1, stride=1):
+    # refer to https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html
+    a = l_in + 2 * padding - dilation * (kernel_size - 1) - 1
+    l_out = math.floor(a / stride + 1)
+    return l_out * channels
+
+
+class XMLCNN(BaseModel):
+    def __init__(self, config, embed_vecs):
+        super(XMLCNN, self).__init__(config, embed_vecs)
+        assert config.fixed_length is True
+
+        self.filter_sizes = config.filter_sizes
+        emb_dim = embed_vecs.shape[1]
```

Comment:
23 and 24 can be swapped so people can connect 22 and 23 together

Reply:
Solved.